packages/plugins/src/plugins/maker/helper.ts

```diff
@@ -0,0 +1,30 @@
+import { Address, PublicClient } from 'viem';
+import { EvmNetworkIdType } from '@sonarwatch/portfolio-core';
+import { proxyRegAbi } from './abis';
+import { proxyRegAddress } from './constants';
+import { getEvmClient } from '../../utils/clients';
+
+/**
+ * Fetches the proxy address for a given user address.
+ * @param userAddress - The address of the user.
+ * @returns The proxy address associated with the user.
+ */
+export const getProxyAddress = async (
+  userAddress: Address,
+  networkId: EvmNetworkIdType
+): Promise<Address | undefined> => {
+  const client: PublicClient = getEvmClient(networkId);
+  try {
+    const proxyAddress = await client.readContract({
+      address: proxyRegAddress as Address,
+      abi: proxyRegAbi,
+      functionName: 'proxies',
+      args: [userAddress],
+    });
+
+    return proxyAddress as Address;
```

Comment:
And, in a kind of _opposite way_, call `getAddress()` instead of casting ?

```suggestion
    return getAddress(proxyAddress);
```

Reply:
but the address is already checksum, calling that is pointless no? I just removed the casting all together